### PR TITLE
Fix: code blocks in mdx + issues/868

### PR DESCRIPTION
**Pull request by muditjuneja**
This PR fixes the following - 

- Show more button in Code blocks
- Added a prop to enable/disable Show more button in Code blocls
- Added text wrap to avoid horizontal scroll
- Added a new package that is traversing hast and adding a new property named metaData which is being passed as a prop

Should close - https://github.com/carbon-design-system/gatsby-theme-carbon/issues/868

### CODE REVIEW

File: packages/gatsby-theme-carbon/src/components/Code/Code.js
Comment (muditjuneja) [inline packages/gatsby-theme-carbon/src/components/Code/Code.js:22]: With this, by default, we are not doing the "Show more" in code blocks.

**Review by eng618 — CHANGES_REQUESTED** [review]
The changes look really good, the only thing I would like to see is the examples updated to reflect this change. Especially adding an example for the **Show More** prop. This would be in the **packages/example/src/pages/components/code-blocks.mdx** file.

File: packages/gatsby-theme-carbon/gatsby-config.mjs
Comment (muditjuneja) [inline packages/gatsby-theme-carbon/gatsby-config.mjs:20]: Created a rehype package which is adding a new prop of named metaData which is being passed in Code.js.

Here is the dirty package - https://www.npmjs.com/package/rehype-mdx-fenced-code-meta-support

**Comment by eng618** [comment]
Lets add some padding to the show more button so it doesn’t overlap the content.

image

**Review by eng618 — APPROVED** [review]
Approving now, but would be good to add the padding to the show more button before we merge this in.

**Comment by muditjuneja** [comment]
> Lets add some padding to the show more button so it doesn’t overlap the content.
> 
> image

I will do that in another follow up PR. Merging this one now.

**Comment by muditjuneja** [comment]
> > Lets add some padding to the show more button so it doesn’t overlap the content.
> > image
> 
> I will do that in another follow up PR. Merging this one now.

I guess I can't merge. I will make that small change.

**Pull request merged**
